### PR TITLE
Remove use of snippet groups

### PR DESCRIPTION
**Pull request by bitjammer**
These are informal in the latest proposal revision and don't carry
any meaning. It's now intended to group snippets via curation in DocC.

### CODE REVIEW

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by bitjammer** [comment]
@swift-ci Please test

File: IntegrationTests/Tests/SnippetDocumentationGenerationTests.swift [hunk 1]
```diff
@@ -25,6 +25,7 @@ final class SnippetDocumentationGenerationTests: XCTestCase {
         XCTAssertEqual(
             Set(dataDirectoryContents.map(\.lastTwoPathComponents)),
             [
+                "documentation/packagewithsnippets.json",
```
Comment (ethan-kusters) [inline IntegrationTests/Tests/SnippetDocumentationGenerationTests.swift:28]: Is this change caused by this PR? Or maybe an unrelated change in DocC that's just making its way through now?

I'm surprised we now have a top level `packagewithsnippets` page.

Comment (bitjammer) [reply IntegrationTests/Tests/SnippetDocumentationGenerationTests.swift:28]: It might be the change that required https://github.com/apple/swift-docc/pull/108 – I'll land that one first and see if it fixes it.

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by bitjammer** [comment]
@swift-ci Please test

File: IntegrationTests/Tests/SnippetDocumentationGenerationTests.swift (hunk 1 above)
Comment (bitjammer) [reply IntegrationTests/Tests/SnippetDocumentationGenerationTests.swift:31]: As it turns out, I have to change the logic in DocC, but yes we'll have to keep this in until that fix lands.

File: IntegrationTests/Tests/SnippetDocumentationGenerationTests.swift [hunk 2]
```diff
@@ -25,6 +25,10 @@ final class SnippetDocumentationGenerationTests: XCTestCase {
         XCTAssertEqual(
             Set(dataDirectoryContents.map(\.lastTwoPathComponents)),
             [
+                // REMOVEME: "documentation/packagewithsnippets.json"
+                // should disappear once the fix for 
+                // https://github.com/apple/swift-docc/pull/108 is available in CI.
```
Comment (ethan-kusters) [inline IntegrationTests/Tests/SnippetDocumentationGenerationTests.swift:30]: Should this be updated to https://github.com/apple/swift-docc/pull/116? Otherwise, I tested with that branch and the issue seems to be resolved so I think this is good-to-go.

**Review by ethan-kusters — APPROVED** [review]

**Comment by bitjammer** [comment]
@swift-ci Please test
